builtin/v8/market/market_params.go

```diff
@@ -0,0 +1,138 @@
+package market
+
+import (
+	addr "github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-bitfield"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/builtin/v8/util/adt"
+	"github.com/filecoin-project/go-state-types/exitcode"
+	cbg "github.com/whyrusleeping/cbor-gen"
+	"golang.org/x/xerrors"
+)
+
+type WithdrawBalanceParams struct {
+	ProviderOrClientAddress addr.Address
+	Amount                  abi.TokenAmount
+}
+
+type PublishStorageDealsParams struct {
+	Deals []ClientDealProposal
+}
+
+type PublishStorageDealsReturn struct {
+	IDs        []abi.DealID
+	ValidDeals bitfield.BitField
+}
+
+// - Array of sectors rather than just one
+// - Removed SectorStart (which is unknown at call time)
+type VerifyDealsForActivationParams struct {
+	Sectors []SectorDeals
+}
+
+type SectorDeals struct {
+	SectorExpiry abi.ChainEpoch
+	DealIDs      []abi.DealID
+}
+
+// - Array of sectors weights
+type VerifyDealsForActivationReturn struct {
+	Sectors []SectorWeights
+}
+
+type SectorWeights struct {
+	DealSpace          uint64         // Total space in bytes of submitted deals.
+	DealWeight         abi.DealWeight // Total space*time of submitted deals.
+	VerifiedDealWeight abi.DealWeight // Total space*time of submitted verified deals.
+}
+
+type ActivateDealsParams struct {
+	DealIDs      []abi.DealID
+	SectorExpiry abi.ChainEpoch
+}
+
+type SectorDataSpec struct {
+	DealIDs    []abi.DealID
+	SectorType abi.RegisteredSealProof
+}
+
+type ComputeDataCommitmentParams struct {
+	Inputs []*SectorDataSpec
+}
+
+type ComputeDataCommitmentReturn struct {
+	CommDs []cbg.CborCid
+}
+
+type OnMinerSectorsTerminateParams struct {
+	Epoch   abi.ChainEpoch
+	DealIDs []abi.DealID
+}
+
+// Validates a collection of deal dealProposals for activation, and returns their combined weight,
+// split into regular deal weight and verified deal weight.
+func ValidateDealsForActivation(
```

Comment:
I can see why you retained this, but I'm not sure it belongs here. This is really just actor logic that happened to be the state file for some reason. While maintaining two language versions of the state schemas is reasonable, maintaining duplicate application logic is not.

This was exported for use by nodes, but in in the future a better approach to things like this will be to call a readonly/view method on an actor locally in order to use its logic. The market actor may even become non-builtin in the future, other market actors certainly won't be.

I don't want to block on this, but this approach won't be sustainable.

This belongs in state, rather than params, I think.

Reply:
Yeah, you're quite that this is going to be unsustainable / will eventually diverge from the real source of truth (builtin actors). We'll open an issue for it.

Agreed that it should be in state, not params, though. Making that change now.